linera-service/src/config.rs

```diff
@@ -263,23 +265,42 @@ impl WalletState {
             };
             Ok(Self {
                 inner,
+                wallet_path: path.into(),
                 _lock: file_lock,
             })
         } else {
             let inner = serde_json::from_reader(reader)?;
             Ok(Self {
                 inner,
+                wallet_path: path.into(),
                 _lock: file_lock,
             })
         }
     }
 
+    /// Writes the wallet to disk.
+    ///
+    /// The contents of the wallet need to be over-written completely, so
+    /// a temporary file is created as a backup in case serialization fails.
+    ///
+    /// The temporary fail is then deleted once we successfully save the
```

Comment:
```suggestion
    /// The temporary file is then deleted once we successfully save the
```

(Also, it's not just about serialization, is it? If that were the case, we could serialize it in memory and then overwrite the file. The reason we need the backup file approach is that the data would be lost if a crash happened while _writing_ the new wallet contents.)